soe/resolvers.go

```diff
@@ -0,0 +1,61 @@
+package soe
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/hamba/avro/v2"
+)
+
+// SchemaResolver implementations are expected to be callable for every record,
+// so should cache expensive schema lookups.
+type SchemaResolver interface {
+	GetSchema(ctx context.Context, fingerprint []byte) (avro.Schema, error)
+}
+
+// ErrUnknownSchema is an API error that must be returned if the requested
+// schema is not known to a resolver.
+var ErrUnknownSchema = errors.New("unknown schema")
+
+// MemorySchemaRegistry is an in-memory cache, intended for testing. It is not
+// safe for multithreaded use.
+type MemorySchemaRegistry struct {
+	schemas map[uint64]avro.Schema
```

Comment:
You should consider a Mutex to protect this or a `SyncMap`. I do not this it can be guaranteed that access to this would be single threaded.

Reply:
Ah, I only envisioned this for tests (hence the `intended for testing` comment). Maybe I should either generalize it for multithreaded use, or move it to a test-internal package.

Reply:
If it is just for tests, it should be in the tests IMO.

Reply:
It can be useful to have a basic store available out of the box, so I switched it to `sync.Map`